
# HG changeset patch
# User John H. Robinson, IV <jaqque@sbih.org>
# Date 1229205226 28800
# Node ID ded22ccbb04f0893e7d33055066674c9a3116bd8
# Parent ff96eb3a10c98136aa55acf0caff18b69ca48ae0
xinerama switchpanel corruption fix

This is a small patch to fix an issue with the switchpanel and a large number
of windows which happens only on a Xinerama setup.

When the number of open windows is so large that displaying all of them would
cause the switchpanel to be too wide for the screen, the panel is supposed to
shrink and scroll to accomodate them all.

In Window Maker 0.92.0 this works for single head displays but not for
Xinerama. The panel extends to the next head and gets garbled. This patch fixes
the issue by correctly constraining the panel to the head with the cursor.

Submitted by: Gilbert Ashley <amigo@ibiblio.org>
Author: Iain Patterson <wm@iain.cx>

--- a/src/switchpanel.c	Sat Dec 13 13:50:14 2008 -0800
+++ b/src/switchpanel.c	Sat Dec 13 13:53:46 2008 -0800
@@ -464,7 +464,7 @@
     iconsThatFitCount= count;
 
     if (width > rect.size.width) {
-        iconsThatFitCount = (WMScreenWidth(scr->wmscreen)-SCREEN_BORDER_SPACING)/ICON_TILE_SIZE;
+        iconsThatFitCount = (rect.size.width-SCREEN_BORDER_SPACING)/ICON_TILE_SIZE;
         width= iconsThatFitCount*ICON_TILE_SIZE;
     }
     

